### PR TITLE
DYN-7154: missing thumbnail sample files

### PR DESCRIPTION
**Pull request by dnenov**
### Purpose

Addressing this `jira` issue: https://jira.autodesk.com/browse/DYN-7154 `Missing thumbnails for sample files in DynamoCore 3.2 new home page`. To be consumed with the HomePage app changes in this PR https://github.com/DynamoDS/DynamoHome/pull/23

### Changes

Screenshot 2024-06-18 170925

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB

### Release Notes

- added the missing properties to the sample files by inheriting from the already existing StartPageListItem
- refactored repetitive code
- tests added
- api updated

### Reviewers

@QilongTang 

### FYIs

@Amoursol

### CODE REVIEW

File: src/DynamoCoreWpf/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -1465,6 +1466,9 @@ Dynamo.UI.Controls.StartPageListItem.DateModified.set -> void
 Dynamo.UI.Controls.StartPageListItem.Description.get -> string
 Dynamo.UI.Controls.StartPageListItem.Icon.get -> System.Windows.Media.ImageSource
 Dynamo.UI.Controls.StartPageListItem.IconVisibility.get -> System.Windows.Visibility
+Dynamo.UI.Controls.StartPageListItem.LoadBitmapImage(string iconPath) -> System.Windows.Media.Imaging.BitmapImage
```
Comment (reddyashish) [inline src/DynamoCoreWpf/PublicAPI.Unshipped.txt:1469]: @QilongTang is it needed to add protected API's to this file?

Comment (QilongTang) [reply src/DynamoCoreWpf/PublicAPI.Unshipped.txt:1469]: I dont think so

Comment (dnenov) [reply src/DynamoCoreWpf/PublicAPI.Unshipped.txt:1469]: I was fixing the errors in VisualStudio. I can remove if not needed.

Comment (dnenov) [reply src/DynamoCoreWpf/PublicAPI.Unshipped.txt]: Removed the protected properties from the API.

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
@dnenov You are missing the change to update the dynamo home version in DynamoCoreWpf.csproj.

**Comment by QilongTang** [comment]
> @dnenov You are missing the change to update the dynamo home version in DynamoCoreWpf.csproj.

yeah, you can refer to https://github.com/DynamoDS/Dynamo/pull/15316/files to define updated version

**Review by reddyashish — APPROVED** [review]

**Comment by QilongTang** [comment]
Thanks @reddyashish @dnenov for the quick turn around

**Review by QilongTang — APPROVED** [review]

**Comment by reddyashish** [comment]
Passed: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16207/

**Pull request merged**